torch/_inductor/codegen/cpp.py

```diff
@@ -1109,6 +1109,21 @@ def eq(x, y):
         assert isinstance(V.kernel, CppVecKernel)
         assert isinstance(x, CppCSEVariable)
         assert x.dtype is not None
+        if (
+            x.dtype in [torch.int32, torch.int64]
+            and y.dtype in [torch.int32, torch.int64]
+            and x.dtype != y.dtype
+        ):
+            x = (
+                f"at::vec::convert<int64_t,2,int32_t,1>({x})"
+                if x.dtype == torch.int32
+                else x
+            )
+            y = (
+                f"at::vec::convert<int64_t,2,int32_t,1>({y})"
+                if y.dtype == torch.int32
+                else y
+            )
```

Comment:
Suggest to move this type promotion to a common place. See this: https://github.com/pytorch/pytorch/pull/122961/files#diff-5ab7b0235e2076a5fc6629ba0b109208940f5b94f5c13babc3e0f87cf4fcec82R1055